packages/backend/src/tokens/interstitialRule.ts

```diff
@@ -75,22 +72,20 @@ export const potentialFirstLoadInDevWhenUATMissing: InterstitialRule = options =
  * It is expected that a primary app will trigger a redirect back to the satellite app.
  */
 export const potentialRequestAfterSignInOrOutFromClerkHostedUiInDev: InterstitialRule = options => {
-  const { apiKey, secretKey, referrer, host, forwardedHost, forwardedProto } = options;
+  const { secretKey, referrer, host, forwardedHost, forwardedProto } = options;
   const crossOriginReferrer =
     referrer && checkCrossOrigin({ originURL: new URL(referrer), host, forwardedHost, forwardedProto });
-  const key = secretKey || apiKey || '';
 
-  if (isDevelopmentFromApiKey(key) && crossOriginReferrer) {
+  if (isDevelopmentFromApiKey(secretKey as string) && crossOriginReferrer) {
```

Comment:
why do we need to `as string` here? is it possible for secretKey to be undefined here? If yes, can isDevelopmentFromApiKey handle it?

Reply:
you are correct. i will drop the usage of the `as string` for these cases since it will cause a `Cannot read properties of undefined (reading 'startsWith') ` error if no secretKey is available. 
When the `authenticateRequest` is executed, this above error will never be seen because it's `secretKey` existence is expected and asserted being this piece of code is being invoked.

fix commit: https://github.com/clerk/javascript/pull/1899/commits/19cbb65dc7cd9d4c6411cee8feeef35df1d6e1db